File: SearchAlgorithmsImpl/LinearSearch.java

public class LinearSearch {

	LinearSearch() {}
	
	public int search(int[] arr, int target) {

		int i = 0;
		for (int ele: arr) {
			if (ele == target) {
				return i;
			}
			i++;
		}

		return 0;
	}

	public static void main (String[] args) {

		int[] arr = {1, 3, 4, 12, 34};

		LinearSearch linearSearch = new LinearSearch();

		System.out.println("Element found at: " + linearSearch.search(arr, 4) + " index.");
	}
}

File: SearchAlgorithmsImpl/LinearSearch.cpp
#include <iostream>
#include <vector>

using namespace std; 

class LinearSearch {

	public:
		void search(const vector<int> &vec, int target) {

			for (size_t i = 0; i < vec.size(); i++) {
				if (vec[i] == target) {
					cout << "Element found at index: " << i << endl;
					return; 
				}
			}

			cout << "Element not found." << endl;
		}
	
};

int main() {
	vector<int> vec = {3, 2, 54, 1, 14, 32};

	
	for (size_t i = 0; i < vec.size(); i++) {
		cout << vec[i] << " ";
	}	

	cout << "\nLooking for " << 14 << "\n";
	
	LinearSearch linearSearch;
	linearSearch.search(vec, 14);
	
	
}
